src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs

```diff
@@ -65,7 +67,7 @@ public class ApiController : Controller
             _databaseProviders = databaseProviders;
             _currentShellSettings = currentShellSettings;
             _notifier = notifier;
-
+            _emailAddressValidator = emailAddressValidator ?? throw new ArgumentNullException(nameof(emailAddressValidator));
```

Comment:
Don't throw exceptions in constructors. Here the DI will fail if the dependency is null

Reply:
I think throwing exception is good to ensure the services should be registered in DI, otherwise weird things may happen. Also it good for testability